<launch>
  <!-- Set this to your camera's name -->
  <arg name="camera_name" value="creative_cam" />

    <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
      <param name="width" type="int" value="1280" />
      <param name="height" type="int" value="720" />
      <param name="fps" type="int" value="30" />
      <param name="format" value="jpeg" />
      <param name="auto_focus" type="bool" value="False" />
      <param name="device" type="string" value="/dev/video0" />
      <param name="camera_info_url" type="string" value="package://uvc_camera/calibration/ost.yaml" />
    </node>


  <include file="/home/sergio/Downloads/3pi_gaming/configurations/config.launch"/>
  <include file="$(find controls)/controls.launch"/>
  <include file="$(find projector_calib)/p_calib.launch"/>

  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
  <!-- Remap topic required by the node to custom topics -->
  <param name="image_transport" value="compressed" />
  <remap from="image_rect" to="image_raw" />
  <remap from="camera_info" to="camera_info"/>

  <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
  <param name="tag_family" type="str" value="36h11" />

  <!-- Enable projected optical measurements for more accurate tag transformations -->
  <!-- This exists for backwards compatability and should be left true for new setups -->
  <param name="projected_optics" type="bool" value="true" />
  <param name="use_cam_info" type="bool" value="true" />

  <!-- Describe the tags -->
  <rosparam param="tag_descriptions">[
    {id: 0, size: 0.077, frame_id: tag_0},
    {id: 1, size: 0.077, frame_id: tag_1},
    {id: 2, size: 0.077, frame_id: tag_2},
    {id: 3, size: 0.077, frame_id: tag_3},
    {id: 4, size: 0.077, frame_id: tag_4},
    {id: 5, size: 0.077, frame_id: tag_5},
    {id: 6, size: 0.077, frame_id: tag_6},
    {id: 7, size: 0.077, frame_id: tag_7},
    {id: 8, size: 0.077, frame_id: tag_8},
    {id: 9, size: 0.077, frame_id: tag_9},
    {id: 10, size: 0.077, frame_id: tag_10},
    {id: 11, size: 0.077, frame_id: tag_11},
    {id: 12, size: 0.077, frame_id: tag_12},
    {id: 13, size: 0.077, frame_id: tag_13},
    {id: 14, size: 0.077, frame_id: tag_14},
    {id: 15, size: 0.077, frame_id: tag_15},
    {id: 16, size: 0.077, frame_id: tag_16},
    {id: 17, size: 0.077, frame_id: tag_17},
    {id: 18, size: 0.077, frame_id: tag_18},
    {id: 19, size: 0.077, frame_id: tag_19},
    {id: 20, size: 0.077, frame_id: tag_20},
    {id: 21, size: 0.077, frame_id: tag_21},
    {id: 22, size: 0.077, frame_id: tag_22},
    {id: 23, size: 0.077, frame_id: tag_23},
    {id: 24, size: 0.077, frame_id: tag_24},
    {id: 25, size: 0.077, frame_id: tag_25},
    {id: 26, size: 0.077, frame_id: tag_26},
    {id: 27, size: 0.077, frame_id: tag_27},
    {id: 28, size: 0.077, frame_id: tag_28},
    {id: 29, size: 0.077, frame_id: tag_29},
    {id: 30, size: 0.077, frame_id: tag_30},
    {id: 31, size: 0.077, frame_id: tag_31},
    {id: 32, size: 0.077, frame_id: tag_32},
    {id: 33, size: 0.077, frame_id: tag_33},
    {id: 34, size: 0.077, frame_id: tag_34},
    {id: 35, size: 0.077, frame_id: tag_35},
    {id: 36, size: 0.077, frame_id: tag_36},
    {id: 37, size: 0.077, frame_id: tag_37}]

  </rosparam>
  </node>

  <node pkg="ai" type="ai" name="ai" output="screen">
  </node>

  <include file="$(find gazebo_ros)/launch/empty_world.launch"/> 


</launch>
